src/Datadog.Trace/Tracer.cs

```diff
@@ -217,10 +217,48 @@ public Span StartSpan(string operationName, ISpanContext parent = null, string s
                 span.SetTag(Tags.Env, env);
             }
 
+            // Apply any global tags
+            if (((TraceContext)traceContext).Tracer.Settings.GlobalTags.Count > 0)
+            {
+                foreach (var entry in ((TraceContext)traceContext).Tracer.Settings.GlobalTags)
+                {
+                    span.SetTag(entry.Key, entry.Value);
+                }
+            }
+
             traceContext.AddSpan(span);
             return span;
         }
 
+        /// <summary>
+        /// Adds the <paramref name="key"/>, <paramref name="value"/> pair as a global tag to be applied to all <see cref="Span"/>s.
+        /// </summary>
+        /// <remarks><paramref name="key"/> is whitespace-trimmed before being added.</remarks>
+        /// <param name="key">The global tag key</param>
+        /// <param name="value">The global tag value</param>
+        public void AddGlobalTag(string key, string value)
+        {
+            if (string.IsNullOrWhiteSpace(key))
+            {
+                Log.Warning("Cannot create a global tag with an empty key.");
+                return;
+            }
+
+            Settings.GlobalTags[key.Trim()] = value;
+        }
+
+        /// <summary>
+        /// Adds each <see cref="KeyValuePair{TKey, TValue}"/> in <paramref name="tags"/> as global tags.
+        /// </summary>
+        /// <param name="tags"><see cref="KeyValuePair{TKey, TValue}"/>s to be added as global tags.</param>
+        public void AddGlobalTags(IEnumerable<KeyValuePair<string, string>> tags)
```

Comment:
Perhaps we can leave this method as internal or private? I'm a little hesitant to open up the public API too quickly and this seems to be more of a convenience method.

Reply:
We decided to axe the APIs and just go with config. But the original Trello ticket was for creating an API, at least in name.